### PR TITLE
add toolchain build/push and install scripts

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: hack/toolchain_build_push.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# parse input
+BRANCH=${BRANCH:-pubviewer-mvp}
+BUILDER=${BUILDER:-docker}
+TAG=${1:-e2etest}
+
+export QUAY_NAMESPACE=${QUAY_NAMESPACE:-konflux-workspaces}
+
+# create a temporary direction
+f=$(mktemp --directory /tmp/toolchain.XXXX)
+cd "$f"
```
Comment (sadlerap) [inline hack/toolchain_build_push.sh:14]:
```suggestion
cd "${f}"
```

File: hack/toolchain_build_push.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+git clone --depth 2 --branch "${BRANCH}" https://github.com/filariow/host-operator.git
+git clone --depth 1 --branch "${BRANCH}" https://github.com/filariow/toolchain-common.git
+git clone --depth 1 --branch "${BRANCH}" https://github.com/filariow/toolchain-api.git api
+git clone --depth 1 --branch "${BRANCH}" https://github.com/filariow/registration-service
+git clone --depth 1 --branch "${BRANCH}" https://github.com/filariow/toolchain-e2e
+
+# build and publish images
+make -C member-operator run-cicd-script \
+  SCRIPT_PATH=scripts/ci/manage-member-operator.sh \
+  SCRIPT_PARAMS="-po true -io false -mn toolchain-member-operator -qn $QUAY_NAMESPACE -ds $TAG -dl false -mr ./"
```
Comment (sadlerap) [inline hack/toolchain_build_push.sh:27]:
```suggestion
  SCRIPT_PARAMS="-po true -io false -mn toolchain-member-operator -qn ${QUAY_NAMESPACE} -ds ${TAG} -dl false -mr ./"
```

File: hack/toolchain_build_push.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+make -C host-operator run-cicd-script \
+  SCRIPT_PATH=scripts/ci/manage-host-operator.sh \
+  SCRIPT_PARAMS="-po true -io false -hn toolchain-host-operator -qn $QUAY_NAMESPACE -ds $TAG -dl false -hr ./ -rr ../registration-service"
```
Comment (sadlerap) [inline hack/toolchain_build_push.sh:31]:
```suggestion
  SCRIPT_PARAMS="-po true -io false -hn toolchain-host-operator -qn ${QUAY_NAMESPACE} -ds ${TAG} -dl false -hr ./ -rr ../registration-service"
```

File: hack/toolchain_install.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+set -e
+
+# parse input
+BRANCH=${BRANCH:-pubviewer-mvp}
+export QUAY_NAMESPACE=${QUAY_NAMESPACE:-workspaces}
+TAG=${1:-e2etest}
+
+# create a temporary direction
+f=$(mktemp --directory /tmp/toolchain.XXXX)
+cd "$f"
```
Comment (sadlerap) [inline hack/toolchain_install.sh:12]:
```suggestion
cd "${f}"
```

File: hack/toolchain_install.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+git clone --depth 2 --branch "${BRANCH}" https://github.com/filariow/registration-service
+
+# deploy
+( 
+  kubectl create namespace toolchain-host-operator --dry-run=client --output=yaml | \
+    kubectl apply -f -
+  kubectl create namespace toolchain-member-operator --dry-run=client --output=yaml | \
+    kubectl apply -f -
+
+  cd "$f/toolchain-e2e"
```
Comment (sadlerap) [inline hack/toolchain_install.sh:29]:
```suggestion
  cd "${f}/toolchain-e2e"
```

File: hack/toolchain_install.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -e
```
Comment (sadlerap) [inline hack/toolchain_install.sh:3]: Maybe add this?
```suggestion
set -e
set -o pipefail
```

File: hack/toolchain_install.sh [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+# deploy
+( 
+  kubectl create namespace toolchain-host-operator --dry-run=client --output=yaml | \
+    kubectl apply -f -
+  kubectl create namespace toolchain-member-operator --dry-run=client --output=yaml | \
+    kubectl apply -f -
+
+  cd "$f/toolchain-e2e"
+
+  make dev-deploy-e2e-local PUBLISH_OPERATOR=false DATE_SUFFIX="$TAG" DEPLOY_LATEST=false
```
Comment (sadlerap) [inline hack/toolchain_install.sh:31]:
```suggestion
  make dev-deploy-e2e-local PUBLISH_OPERATOR=false DATE_SUFFIX="${TAG}" DEPLOY_LATEST=false
```

**Review by sadlerap — CHANGES_REQUESTED** [review]
Had a bunch of suggestions from running `shellcheck` locally, not sure why CI didn't pick up on them.  Let me look into it.

File: hack/toolchain_install.sh [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+git clone --depth 2 --branch "${BRANCH}" https://github.com/filariow/toolchain-common.git
+git clone --depth 2 --branch "${BRANCH}" https://github.com/filariow/toolchain-api.git api
+git clone --depth 2 --branch "${BRANCH}" https://github.com/filariow/registration-service
+
+# deploy
+( 
+  kubectl create namespace toolchain-host-operator --dry-run=client --output=yaml | \
+    kubectl apply -f -
+  kubectl create namespace toolchain-member-operator --dry-run=client --output=yaml | \
+    kubectl apply -f -
```
Comment (sadlerap) [inline hack/toolchain_install.sh:27]: We have a mix of `kubectl` and `oc` invocations, even though I think everything is covered by `kubectl` directly.  I'd prefer we use one tool and stick to it.

**Comment by filariow** [comment]
> Had a bunch of suggestions from running `shellcheck` locally, not sure why CI didn't pick up on them. Let me look into it.

On my local, shellcheck v0.9.0 is not providing any suggestion. Which version of shellcheck are you using?
I think the check that suggests changes like `$f` -> `${f}` is an optional one. Did you enable some optional checks?

File: hack/toolchain_install.sh (hunk 6 above)
Comment (filariow) [reply hack/toolchain_install.sh:3]: ATM it's not needed as no piped command is there. shellcheck `check-extra-masked-returns` should prevent us from adding a piped command without adding `-o pipefail` or splitting the command. So we can add this in future if needed

**Comment by sadlerap** [comment]
Yes, I've largely been running it locally with the `-o all` flag (local version is `0.9.0`).  It might be a bit overkill though...

**Comment by sadlerap** [comment]
I'll go take out `-o all` from `.github/workflows/security.yaml`, since I explicitly set it there.  Now that we have a `.shellcheckrc`, it should be safe to remove.

**Comment by sadlerap** [comment]
Filed #67.

**Review by sadlerap — APPROVED** [review]

**Pull request merged**